compiler/damlc/daml-package-config/src/DA/Daml/Package/Config.hs

```diff
@@ -66,6 +68,16 @@ parseProjectConfig project = do
     pSdkVersion <- queryProjectConfigRequired ["sdk-version"] project
     Right PackageConfigFields {..}
 
+checkPkgConfig :: PackageConfigFields -> [String]
+checkPkgConfig PackageConfigFields {pName, pVersion} =
+  [ "WARNING: Package names should have the format ^[a-zA-Z]([a-zA-Z\\-])*$. You may be able to compile packages with different formats, but you will not be able to use them as dependencies in other projects. Unsupported package names may start causing compilation errors without warning."
+  | not $ LF.unPackageName pName =~ ("^[a-zA-Z]([a-zA-Z\\-])*$" :: T.Text)
+  ] ++
+  [ "WARNING: Package versions should have the format ^[0-9](\\.[0-9])*$. You may be able to compile packages with different formats, but you will not be able to use them as dependencies in other projects. Unsupported package versions may start causing compilation errors without warning."
+  | Just version <- [pVersion]
+  , not $ LF.unPackageVersion version =~ ("^[0-9](\\.[0-9])*$" :: T.Text)
```

Comment:
I think we should try to allow only versions that are compatible with both Haskell's PVP and the more common SemVer. AFAICT this would be the regex `[1-9][0-9]*(\\.[1-9][0-9]*(\\.[1-9][0-9]*)?)?`, but you could also just use the SemVer package (that we already depend on in daml-assistant for SDK versions ... even though it's not my favorite), and make sure the "prerelease tags" are empty.

[If we want to support full SemVer versions at some point, we could strip out the "prerelease tags" when feeding the version to GHC.]

Reply:
changed regex now to what sofia proposes.